app/scripts/components/exploration/components/datasets/data-layer-card.tsx

```diff
@@ -123,16 +116,18 @@ const LegendColorMapTrigger = styled.div`
 export default function DataLayerCard(props: CardProps) {
   const {
     dataset,
-    datasetAtom,
     isVisible,
     setVisible,
     colorMap,
     setColorMap,
     colorMapScale,
     setColorMapScale,
-    datasetLegend,
     onClickLayerInfo
   } = props;
+
+  const datasetAtom = useTimelineDatasetAtom(dataset.data.id);
```

Comment:
We would want to get rid of the jotai atom in this component. Should we pass the `dataset.data.id` directly to `LayerMenuOptions` and delegate the fetching?

Reply:
@hanbyul-here PR https://github.com/NASA-IMPACT/veda-ui/pull/1809 should address this.